### ISSUE TITLE
Clean up the preview list / improve usability

### ISSUE THREAD

**Issue by minimeee** [mtkennerly/ludusavi#159]
### What's your idea?

Hi! I really like Ludusavi but imho working with the preview list of saves is inconvenient when it doesn't fit on the screen (and this happens pretty fast). I don't like that I have to scroll each time to check if all games I want to back up are actually selected. Or that I have to scroll in order to check their state. I'm usually playing 2-3 games at a time and they quickly get lost in the long list of game saves. This is especially true on the small Steam Deck screen.

My first idea was: hide the games you don't want to back up anymore (but also don't want to delete the local saves). But actually it would be sufficient to have the selected games (those that you want to back up, because you're currently playing them) **always at the top of the list**, so I can see the current state at a glance. 

However, removing games from the list, so their saves are not even checked, may also be nice if it would speed up the preview function. But then you'd need a way to make them visible again...

**Comment by mtkennerly** [comment]
I think that makes sense to have an option to sort by: selected -> deselected. Another option that could help would be to sort by status: new -> changed -> unchanged -> unscanned. For you personally, is it important to have them sorted to the top even before you've previewed them? As in, do you usually preview specific games or just preview everything at once?

**Comment by minimeee** [comment]
> Another option that could help would be to sort by status: new -> changed -> unchanged -> unscanned

That could be helpful in some cases. For me it's not that important. I have saves from old games on my disk that Ludusavi always marks as "new" because I never did a backup, but I don't care about them either. Maybe I should delete them from the disk... anyway, seeing them on top doesn't make sense for me personally. I'd rather ignore them completely.

> For you personally, is it important to have them sorted to the top even before you've previewed them?

Yes, selected games should always be on top, even before preview. That would also be more consistent, because otherwise the order would change when you do "preview", which would be confusing. In general, for me selecting games is a way to say "these are currently important to me, please prioritize them". Thus, showing them on top just makes sense imo.

> As in, do you usually preview specific games or just preview everything at once?

Tbh, I didn't even know that was possible :relaxed: I've checked again and indeed the option is there in the `...` menu. But imho that's too hidden and cumbersome to use it regularly (especially on the Steam Deck). 

Having the option to do "preview" only for selected games would make more sense (and improve performance). This behaviour would also be consistent with how "backup" works (which only backs up selected games). In general, it would basically be the same as having the option to "hide" or "ignore" games, but they would remain visible (just not processed at all). Not sure which option is better, though...

Basically, this discussion boils down to the question, "what does selecting individual games mean?". Currently, it only affects the backup function, but not the preview, which is somewhat inconsistent. To me, it has a more general meaning, as I (hopefully) made clear above.

**Comment by minimeee** [comment]
I had another idea that I'd like to share: you could separate the games list in two areas: one containing all games detected and one containing only the selected or "enabled" games, that the user actually wants to back up (i. e. those he's currently playing). The areas could be implemented as two tabs.

The "enabled" list / area / tab could basically be identical to the current list, but _all_ games are previewed and backed up when using the respective button at the top. The `...` menu could contain a `disable` action (to remove the game from that list) and the checkbox column could be deleted. Also, a column of little `back up` buttons could replace the current checkboxes, so you can back up a single entry with a single click (without having to use the `...` menu).

The "all" list could be similar, but with an `enable` entry in the `...` (and maybe also a `disable`, since the list contains _all_ games). Also, you could replace the checkbox columns with little `enable` / `disable` buttons that change depending on the state.

The "restore" mode could remain as it is now and show all games that have a backup. Since you usually restore only a single game at once, the checkbox column could be replaced by `restore` buttons.

If you'd actually replace the checkbox columns in all modes, as described above, you'd keep the UI consistent, i. e. all views have a button column with the most important action for that mode.

OK, that became longer that I intended. Maybe I should have drawn a mockup instead :sweat_smile:

**Comment by minimeee** [comment]
> this my be really subjective, but remove the 3 dot and make it a right click option, 

Please don't. Right click menus are very bad for touch input and also not very intuitive for desktop use, since they are hidden per definition.

> you can make it known by a small help section,

Or you can keep the `...` menu, which makes it clear that more options are available without needing explanation :wink: It's better imo, although a bit small for the Steam Deck screen. That's why I suggested to replace the checkbox with a button for each game that triggers the "main action" for the respective mode (i. e. back up, restore, etc.).

> add remove from list option to never see the game name if the user requires to, and have a list in options, something like "ignored" to remove the game from that list again if the user changed its mind

I agree with this part, though.

**Comment by Triplesweet** [comment]
If we think about touch devices yes having the 3 dot makes more sense

**Comment by mtkennerly** [comment]
The next release will include a few filters for the scan results:

> https://user-images.githubusercontent.com/4934192/228530637-9c06d84b-a52e-4739-a07e-7f1c4c6ce8bb.mp4
